ext/drcovlib/modules.c

```diff
@@ -174,11 +176,12 @@ event_module_load(void *drcontext, const module_data_t *data, bool loaded)
                 }
             }
 #endif
-            break;
+            /* We may just see a sub module, so we cannot stop here. We need iterate
+             * all entries to update all sub modules of the reloaded module.
```

Comment:
The first segment entry should match as all segments share the same module_data_t pointer.  Why didn't the first one match?